src/components/Gesture.svelte

```diff
@@ -72,33 +71,36 @@
   let isThisRecording = false;
   let showCountdown = false;
   let countdownIdx = 0;
-  $: currCountdownConfig = countdownConfigs[countdownIdx];
```

Comment:
Haven't quite understood why inlining `currCountdownConfig` fixes the bug (where the second time button B is pressed the recording doesn't work), but found the following: 

I tried adding `$: console.log("countdownIdx", countdownIdx)` before this line. Interestingly, `countdownIdx` doesn't get logged the second time button B gets pressed. 

But, if I move the reactive declaration(s) to [here](https://github.com/microbit-foundation/ml-trainer/blob/fea374478d5cb7af929e040a9c9eb63c04030cb7/src/components/Gesture.svelte#L265), which is after the `triggerButtonsClicked($buttonPressed)` is called, the `currCountdownConfig` reacts as expected, and the `countdownIdx` gets logged consistently when button B gets pressed.

It makes me think there is some magic going on regarding the ordering of these reactive statements. I tried to reproduce the issue via a playground script, but was unable to, which makes me think I don't understand this fully.